content/docs/tasks/security/rbac-groups/index.md

```diff
@@ -0,0 +1,185 @@
+
+{{< text bash >}}
+$ kubectl exec $(kubectl get pod -l app=sleep -n $NS -o jsonpath={.items..metadata.name}) -c sleep -n $NS -- curl http://httpbin.$NS:8000/ip -s -o /dev/null -w "%{http_code}\n"
+401
+{{< /text >}}
+
+### Configure groups-based authorization
+
+This section will create a policy to authorize the access to the `httpbin` service if the requests are
+originated from specific groups.
+
+First, Istio RBAC should be enabled for the namespace by running the following command:
+
+{{< text bash >}}
+$ cat <<EOF | kubectl apply -n $NS -f -
+apiVersion: "rbac.istio.io/v1alpha1"
+kind: RbacConfig
+metadata:
+  name: default
+spec:
+  mode: 'ON_WITH_INCLUSION'
+  inclusion:
+    namespaces: ["rbac-groups-test-ns"]
+EOF
+{{< /text >}}
+
+As there may be some delays due to caching and other propagation overhead, wait a moment for the newly
+defined RBAC policy to become effective. Run the following command to verify that the curl connection
+to the httpbin service is rejected after the RBAC policy has become effective.
+
+{{< text bash >}}
+$ kubectl exec $(kubectl get pod -l app=sleep -n $NS -o jsonpath={.items..metadata.name}) -c sleep -n $NS -- curl http://httpbin.$NS:8000/ip -s -o /dev/null -w "%{http_code}\n" --header "Authorization: Bearer $TOKEN"
+403
+{{< /text >}}
+
+Next, create a `ServiceRole`, called httpbin-viewer, that allows read access to the `httpbin` service.
+
+{{< text bash >}}
+$ cat <<EOF | kubectl apply -n $NS -f -
+apiVersion: "rbac.istio.io/v1alpha1"
+kind: ServiceRole
+metadata:
+  name: httpbin-viewer
+  namespace: rbac-groups-test-ns
+spec:
+  rules:
+  - services: ["httpbin.rbac-groups-test-ns.svc.cluster.local"]
+    methods: ["GET"]
+EOF
+{{< /text >}}
+
+Create a `ServiceRoleBinding` that assigns the `httpbin-viewer` role to users in the `group1`.
+
+{{< text bash >}}
+$ cat <<EOF | kubectl apply -n $NS -f -
+apiVersion: "rbac.istio.io/v1alpha1"
+kind: ServiceRoleBinding
+metadata:
+  name: bind-httpbin-viewer
+  namespace: rbac-groups-test-ns
+spec:
+  subjects:
+  - user: "*"
+    properties:
+      request.auth.claims[groups]: "group1"
+  roleRef:
+    kind: ServiceRole
+    name: "httpbin-viewer"
+EOF
+{{< /text >}}
+
+Wait a moment for the newly defined RBAC policy to become effective. Run the following command to verify
+that the curl connection to the httpbin service succeeds after the RBAC policy has become effective.
+The curl connection to the httpbin service succeeds because its header includes a valid JWT with the
+`groups` claim value [`"group1"`, `"group2"`] containing `group1`.
+
+{{< text bash >}}
+$ kubectl exec $(kubectl get pod -l app=sleep -n $NS -o jsonpath={.items..metadata.name}) -c sleep -n $NS -- curl http://httpbin.$NS:8000/ip -s -o /dev/null -w "%{http_code}\n" --header "Authorization: Bearer $TOKEN"
+200
+{{< /text >}}
+
+## Cleanup
+
+{{< text bash >}}
+$ kubectl delete -f <(istioctl kube-inject -f @samples/httpbin/httpbin.yaml@) -n $NS
```

Comment:
Probably you can just delete the namespace first, then all resources in that namespace will be deleted automatically.

Reply:
I tried "kubectl delete namespace -n $NS" but it only deletes the namespace, not all resources in the namespace.

Reply:
Could you try `kubectl delete namespace $NS` (without the `-n` in the command)?

Reply:
Done: added the cleanup using "kubectl delete namespace $NS".